acord/models/member.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import pydantic
+from acord.bases import Hashable, UserFlags
+from typing import Any, List, Optional
+import datetime
+
+from .user import User
+
+class Member(pydantic.BaseModel, Hashable):
+    """
+    Represents a guild member.
+
+    Attributes
+    ----------
+    user: :class:`acord.models.User`
+          The User object of the member. Not included in MESSAGE_CREATE and MESSAGE_UPDATE events
+    nick: :class:`str`
+          Guild specific nickname of the member
+    avatar: :class:`str`
+          Member\'s guild avatar hash
+    roles: :class:`List[acord.models.Snowflake]`
+          List of role IDs of roles the user has
+    joined_at: :class:`datetime.datetime`
+          The time the user joined the guild
+    premium_since: :class:`datetime.datetime`
+          When the user started boosting the guild
+    deaf: :class:`bool`
+          Whether if the member is deafened in voice channels
+    mute: :class:`bool`
+          Whether if the member is mutes in voice channels
+    pending: :class:`bool`
+          Where if the member is pending verification. Not included (False) in non-GUILD_\* events
+    permissions: :class:`str`
+          Total permissions of the member in the channel. Including overwrites.
+    """
+    # Please check the doc strings before merging @Seniatical
+    # and tell me if i need to add anything else :)
+
+    con: Any # connection object
+
+    user: Optional[User] # not included in MESSAGE_CREATE and MESSAGE_UPDATE events
+
```

Comment:
May need to add on additional vars depending on how we receive the event

Reply:
wdym "additional vars"?

Reply:
guild user belongs to
tho I think that's usually received with `GUILD_CREATE`

Reply:
why isnt it documented if its included in the guild-member object- :moyai: they should have documented it and tell which event its sent in

Reply:
I could do that if i get proper docs links :smugdoggo:

Reply:
> why isnt it documented if its included in the guild-member object- 🗿 they should have documented it and tell which event its sent in

Eh not too sure, you receive members from `GUILD_CREATE` and `GUILD_MEMBER_ADD`

Reply:
the param is called `guild_id` if im not wrong

Reply:
> the param is called `guild_id` if im not wrong

Yes it is.
https://discord.com/developers/docs/topics/gateway#guild-member-add